packages/esm-patient-labs-app/src/lab-orders/add-lab-order/lab-order-form.component.tsx

```diff
@@ -2,18 +2,45 @@ import React, { useCallback, useState } from 'react';
 import classNames from 'classnames';
 import { launchPatientWorkspace, useOrderBasket } from '@openmrs/esm-patient-common-lib';
 import { useLayoutType, useSession } from '@openmrs/esm-framework';
-import { type LabOrderBasketItem, careSettingUuid, prepLabOrderPostData } from '../api';
-import { Button, ButtonSet, Column, ComboBox, Form, Layer, Grid, InlineNotification, TextInput } from '@carbon/react';
+import { careSettingUuid, type LabOrderBasketItem, prepLabOrderPostData } from '../api';
+import {
+  Button,
+  ButtonSet,
+  Column,
+  ComboBox,
+  Form,
+  Layer,
+  Grid,
+  InlineNotification,
+  TextInput,
+  FormLabel,
+} from '@carbon/react';
 import { useTranslation } from 'react-i18next';
 import { priorityOptions } from './lab-order';
-import { type TestType, useTestTypes } from './useTestTypes';
+import { useTestTypes } from './useTestTypes';
 import styles from './lab-order-form.scss';
+import { Controller, useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
 
 export interface LabOrderFormProps {
   initialOrder: LabOrderBasketItem;
   closeWorkspace: () => void;
 }
 
+const labOrderFormSchema = z.object({
+  urgency: z.string({ required_error: 'Priority is required', invalid_type_error: 'Priority is required' }),
+  instructions: z.string().optional(),
+  labReferenceNumber: z.string({
+    required_error: 'Lab reference number is required',
+    invalid_type_error: 'Lab reference number is required',
+  }),
+  testType: z.object(
+    { label: z.string(), conceptUuid: z.string() },
+    { required_error: 'Test type is required', invalid_type_error: 'Test type is required' },
```

Comment:
Hi @usamaidrsk!
All these texts must be translated. If you are using any text inside a hook or component, it should use `t` from `useTranslation`. If outside of the component, please use `translateFrom` with first argument being the moduleName.